os-eap-sso/added/keycloak.sh

```diff
@@ -436,10 +435,8 @@ function read_web_dot_xml {
   local result=
 
   if [ -d "$jarfile" ]; then
-    if [[ -n "$AUTO_DEPLOY_EXPLODED" && "$AUTO_DEPLOY_EXPLODED" == "true" ]] || [[ -n "$JAVA_OPTS_APPEND" && $JAVA_OPTS_APPEND == *"Xdebug"* ]]; then
-      if [ -e "${jarfile}/${filename}" ]; then
+    if [ -e "${jarfile}/${filename}" ]; then
```

Comment:
@iankko I think this change is ok but I wanted to point it out to you in case you thought otherwise.

AIUI what this function does is it provides deployment descriptor content to the configure_subsystem function, which then checks the auth-method, and if it matches KEYCLOAK or SAML, then that factors the deployment into the configuration of the subsystem.

My impression is the point of the removed L439 was to avoid looking into exploded deployments unless some env vars were set in such a way that it was thought that the deployment would get deployed.  IOW ignore deployments that won't be deployed. So removing L439 means there's a chance some deployment gets processed that wouldn't be deployed. Which seems like an edge case.

Even without the CLOUD-2912 problem, the old L439 wasn't quite right as it was not accounting for the case where $jarfile would be deployed because a ${jarfile}.dodeploy marker file exists.  If this overall PR went in but you wanted the L439 check preserved, then that would need to be corrected, since "$AUTO_DEPLOY_EXPLODED" is no longer likely to be "true".

But if ignoring some deployments isn't important, then the change here is simpler.

Reply:
@bstansberry 

Hey Brian,

`My impression is the point of the removed L439 was to avoid looking into exploded deployments unless some env vars were set in such a way that it was thought that the deployment would get deployed. IOW ignore deployments that won't be deployed. So removing L439 means there's a chance some deployment gets processed that wouldn't be deployed. Which seems like an edge case.`

FWICT looking at the code in question, yes, that's probably the explanation. Other PoV / reason why it was writtent, as it wast, might be, we don't want to process only deployments, that will be truly deployment, but rather pragmatical approach - if the file is already expanded in plaintext (exploded), read `web.xml` from it. If it isn't yet, unzip the the archive, and preprocess it. IOW speed-up the `web.xml` retrieval.

`Even without the CLOUD-2912 problem, the old L439 wasn't quite right as it was not accounting for the case where $jarfile would be deployed because a ${jarfile}.dodeploy marker file exists. If this overall PR went in but you wanted the L439 check preserved, then that would need to be corrected, since "$AUTO_DEPLOY_EXPLODED" is no longer likely to be "true".`

Agree, `AUTO_DEPLOY_EXPLODED` shouldn't default to `true` afaict, per:

- [1] https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.1/html-single/configuration_guide/#reference_deploy_scanner_attr or
- [2] https://docs.jboss.org/author/display/AS71/Deployment+Scanner+configuration

From [2] `Setting this to 'true' is not recommended for anything but basic development scenarios, as there is no way to ensure that deployment will not occur in the middle of changes to the content. ` => defaulting to `false` seems reasonable to me.

The added [line in PR](https://github.com/jboss-openshift/cct_module/pull/304/files#diff-fce22065a6ac781b114c39c8eb087de0R157) seems like functional alternative of setting `AUTO_DEPLOY_EXPLODED=true`, without the possibly negative effect of causing re-deployments.

The proposed PR LGTM (from code review PoV).

@ruromero @bstansberry @rcernich @luck3y Given the testing images won't show some regression (see comment from Rob about providing them below), ACK for the change.